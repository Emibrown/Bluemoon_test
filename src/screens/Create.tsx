import React, {useState} from 'react';
import { 
    StyleSheet, 
    Text, 
    View ,
    TouchableOpacity,
    ScrollView,
    TextInput
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {SafeAreaView } from 'react-native-safe-area-context';
import { useDispatch, useSelector } from 'react-redux';
import { scale, verticalScale, moderateScale } from 'react-native-size-matters';
import { appState } from '../utils/interface';
import { colors } from '../theme/colors';
import inventoryService from '../utils/inventoryService';
import { inventoryData } from '../utils/interface';
import uuid from 'react-native-uuid';
import Toast from "react-native-simple-toast";
import { StackScreenProps} from '@react-navigation/stack';

type RootStackParamList = {
  Home: undefined;
  Create: undefined;
  Edit: inventoryData
};

type Props = StackScreenProps<RootStackParamList, 'Create'>;


const Create = ({navigation,route}: Props) => {
  const state = useSelector((state:appState) => state);
  const dispatch = useDispatch();

  const [name,setName] = useState<string>("");
  const [stock,setStock] = useState<number>(0);
  const [price,setPrice] = useState<number>(0);
  const [description,setDescription] = useState<string>("");

  const addInventory = async () => {
    // Name required
    if(!name) return Toast.show('Item name is required');
     // Name is unique
    if(state.inventory.some((el:inventoryData) => el.name == name)) return Toast.show('Item with this name already exist');
    // stock is required
    if(!stock && stock !== 0) return Toast.show('Item stock is required');
    // price is required
    if(!price && price !== 0) return Toast.show('Item price is required');
    //  description is required
    if(!description) return Toast.show('Item description is required');
    // description must have at least three words
    if(description.split(' ').length < 3) return Toast.show('Item description is too short');
    const newInventory: inventoryData = {
      _id:  uuid.v4().toString(),
      name,
      stock,
      price,
      description
    }
    inventoryService.addInventory(newInventory,dispatch)
    .then((resp)=>{
      Toast.show('Item successfully added to inventory');
      navigation.goBack()
    })
    .catch((err)=>{
      Toast.show('Error! Can not add Item to inventory');
    })
  }

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.header}>
        <TouchableOpacity onPress={()=>navigation.goBack()}>
          <Icon name="keyboard-backspace" size={40} />
        </TouchableOpacity>
        <Text style={styles.headerText}>
            Add Inventory
        </Text>
      </View>
      <ScrollView>
        <View style={styles.formGroup}>
          <Text style={styles.label}>
            Name
          </Text>
          <TextInput  
          style={styles.input} 
          onChangeText={(text) => setName(text)}
          />
        </View>

        <View style={styles.formGroup}>
          <Text style={styles.label}>
            Total stock
          </Text>
          <TextInput  
          keyboardType="numeric" 
          style={styles.input}
          value={stock || stock === 0 ? stock+"":""}
          onChangeText={(text) => setStock(parseInt(text))}
          />
        </View>

        <View style={styles.formGroup}>
          <Text style={styles.label}>
            Price
          </Text>
          <TextInput  
          keyboardType="numeric" 
          style={styles.input} 
          value={price || price === 0 ? price+"":""}
          onChangeText={(text) => setPrice(parseInt(text))}
          />
        </View>

        <View style={styles.formGroup}>
          <Text style={styles.label}>
            Description
          </Text>
          <TextInput 
          style={styles.input}
          onChangeText={(text) => setDescription(text)}
          />
        </View>

        <TouchableOpacity onPress={addInventory} style={styles.btn}>
          <Text style={styles.btnText}>
            Add Item
          </Text>
        </TouchableOpacity>
        
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container:{
      flex:1,
      backgroundColor:colors.white
  },
  header:{
      paddingVertical:"2%",
      paddingHorizontal:"5%"
  },
  headerText:{
      fontSize:moderateScale(22),
      fontWeight:"bold",
      color:colors.primary
  },
  formGroup:{
    marginHorizontal:"5%",
    marginTop:"5%"
  },
  label:{
    fontWeight:'400',
    fontSize:moderateScale(16)
  },
  input:{
    backgroundColor:"#e0e0e0",
    height:moderateScale(45),
    paddingLeft:"5%",
  },
  btn:{
    backgroundColor:colors.primary,
    alignItems:"center",
    justifyContent:"center",
    paddingVertical:"4%",
    marginHorizontal:"5%",
    marginTop:"10%"
  },
  btnText:{
    fontSize:moderateScale(16),
  }
});

export default Create;