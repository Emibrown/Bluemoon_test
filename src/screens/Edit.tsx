import React, {useState,useEffect} from 'react';
import { 
    StyleSheet, 
    Text, 
    View ,
    TouchableOpacity,
    ScrollView,
    TextInput,
    Alert
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {SafeAreaView } from 'react-native-safe-area-context';
import { useDispatch, useSelector } from 'react-redux';
import { scale, verticalScale, moderateScale } from 'react-native-size-matters';
import { appState } from '../utils/interface';
import { colors } from '../theme/colors';
import inventoryService from '../utils/inventoryService';
import { inventoryData } from '../utils/interface';
import Toast from "react-native-simple-toast";
import { StackScreenProps} from '@react-navigation/stack';


type RootStackParamList = {
  Home: undefined;
  Create: undefined;
  Edit: inventoryData
};

type Props = StackScreenProps<RootStackParamList, 'Edit'>;


const Edit = ({navigation,route}: Props) => {
  const state = useSelector((state:appState) => state);
  const dispatch = useDispatch();
  const _id = route.params._id
  const [name,setName] = useState<string>(route.params.name);
  const [stock,setStock] = useState<number>(route.params.stock);
  const [price,setPrice] = useState<number>(route.params.price);
  const [description,setDescription] = useState<string>(route.params.description);

  // // Name is unique check
  const uniqueName = () => {
    const found = state.inventory.find((el:inventoryData) => el.name == name);
    if(!found){
      return true
    }else{
      if(found._id == _id){
        return true
      }
    }
    return false
  }

  const EditInventory = async () => {
    // Name required
    if(!name) return Toast.show('Item name is required');
     // Name is unique
    if(!uniqueName()) return Toast.show('Item with this name already exist');
    // stock is required
    if(!stock && stock !== 0) return Toast.show('Item stock is required');
    // price is required
    if(!price && price !== 0) return Toast.show('Item price is required');
    //  description is required
    if(!description) return Toast.show('Item description is required');
    // description must have at least three words
    if(description.split(' ').length < 3) return Toast.show('Item description is too short');
    const newInventory: inventoryData = {
      _id,
      name,
      stock,
      price,
      description
    }
    inventoryService.EditInventory(newInventory,dispatch)
    .then((resp)=>{
      Toast.show('Item successfully updated');
    })
    .catch((err)=>{
      Toast.show('Error! Can not update Item');
    })
  }

  const onDelete = () => {
    Alert.alert(
      "Alert",
      "Are youe sure you want to remove this item from inventory",
      [
        { text: "Yes", style:'destructive', onPress: () => RemoveInventory() },
        {
          text: "No",
          style: "cancel"
        },
      ],
      { cancelable: true }
    );
  }

  const RemoveInventory = async () => {
    inventoryService.RemoveInventory(_id,dispatch)
    .then((resp)=>{
      Toast.show('Item deleted successfully');
      navigation.goBack()
    })
    .catch((err)=>{
      Toast.show('Error! Can not delete Item');
    })
  }



  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.header}>
        <TouchableOpacity onPress={()=>navigation.goBack()}>
          <Icon name="keyboard-backspace" size={40} />
        </TouchableOpacity>
        <Text style={styles.headerText}>
            Edit Inventory
        </Text>
      </View>
      <ScrollView>
        <View style={styles.formGroup}>
          <Text style={styles.label}>
            Name
          </Text>
          <TextInput  
          style={styles.input} 
          value={name}
          onChangeText={(text) => setName(text)}
          />
        </View>

        <View style={styles.formGroup}>
          <Text style={styles.label}>
            Total stock
          </Text>
          <TextInput  
          keyboardType="numeric" 
          style={styles.input}
          value={stock || stock === 0 ?stock.toString():""}
          onChangeText={(text) => setStock(parseInt(text))}
          />
        </View>

        <View style={styles.formGroup}>
          <Text style={styles.label}>
            Price
          </Text>
          <TextInput  
          keyboardType="numeric" 
          style={styles.input} 
          value={price || price === 0 ? price+"":""}
          onChangeText={(text) => setPrice(parseInt(text))}
          />
        </View>

        <View style={styles.formGroup}>
          <Text style={styles.label}>
            Description
          </Text>
          <TextInput 
          style={styles.input}
          value={description}
          onChangeText={(text) => setDescription(text)}
          />
        </View>

        <TouchableOpacity onPress={EditInventory} style={styles.btn}>
          <Text style={styles.btnText}>
            Add Item
          </Text>
        </TouchableOpacity>

        <TouchableOpacity onPress={onDelete} style={{...styles.btn,backgroundColor:"#e57373"}}>
          <Text style={styles.btnText}>
            Delete Item
          </Text>
        </TouchableOpacity>
        
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container:{
      flex:1,
      backgroundColor:colors.white
  },
  header:{
      paddingVertical:"2%",
      paddingHorizontal:"5%"
  },
  headerText:{
      fontSize:moderateScale(22),
      fontWeight:"bold",
      color:colors.primary
  },
  formGroup:{
    marginHorizontal:"5%",
    marginTop:"5%"
  },
  label:{
    fontWeight:'400',
    fontSize:moderateScale(16)
  },
  input:{
    backgroundColor:"#e0e0e0",
    height:moderateScale(45),
    paddingLeft:"5%",
  },
  btn:{
    backgroundColor:colors.primary,
    alignItems:"center",
    justifyContent:"center",
    paddingVertical:"4%",
    marginHorizontal:"5%",
    marginTop:"10%"
  },
  btnText:{
    fontSize:moderateScale(16),
  }
});

export default Edit;