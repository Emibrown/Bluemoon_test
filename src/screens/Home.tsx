import React,{useCallback,useEffect} from 'react';
import { 
    StyleSheet, 
    Text, 
    View,
    StatusBar,
    TouchableOpacity,
    FlatList
} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import {SafeAreaView } from 'react-native-safe-area-context';
import { useDispatch, useSelector } from 'react-redux';
import { scale, verticalScale, moderateScale } from 'react-native-size-matters';
import { appState } from '../utils/interface';
import { colors } from '../theme/colors';
import Loader from '../components/Loader';
import inventoryService from '../utils/inventoryService';
import { inventoryData } from '../utils/interface';
import { ADD_TO_INVENTORY, LOADING_STATUS, UPDATE_INVENTORY } from '../stores/action';
import { StackScreenProps} from '@react-navigation/stack';
import Button from '../components/Button';


type RootStackParamList = {
  Home: undefined;
  Create: undefined;
  Edit: inventoryData
};

type Props = StackScreenProps<RootStackParamList, 'Home'>;



const Home = ({navigation,route}: Props) => {
    const state = useSelector((state:appState) => state);
    const dispatch = useDispatch();

    useEffect( ()=>{
        const bootstrap = async () => {
            const inventory = await inventoryService.getInventory()
            dispatch({ type: UPDATE_INVENTORY,  payload:inventory });
            dispatch({ type: LOADING_STATUS,  payload:false });
        };
        bootstrap();
    },[])

    const keyExtractor = useCallback(
        (item, i) => i.toString(),
        []
    )
    
    const renderItem = useCallback(
        ({ item, index }) => <Button item={item} onPress={()=>navigation.navigate("Edit",item)} />,
        []
    )
    

    return (
        <SafeAreaView testID="2" style={styles.container}>
            <StatusBar 
                // translucent ={true}
                backgroundColor="#509BC7"
                barStyle="dark-content"
                showHideTransition="slide"
            />
            <View  style={styles.header}>
                <Text style={styles.headerText}>
                    Bluemoon
                </Text>
            </View>
            {
                state.loading?(
                    <Loader size={40} />
                ):(
                    state.inventory.length > 0?(
                    <FlatList
                        data={state.inventory}
                        renderItem={renderItem}
                        keyExtractor={keyExtractor}
                        removeClippedSubviews={true}
                        maxToRenderPerBatch={30}
                        updateCellsBatchingPeriod={0}
                        initialNumToRender={50}
                        windowSize={21}
                        onEndReachedThreshold={0.01}
                        style={{
                            paddingVertical:"5%",
                            paddingHorizontal:"5%"
                        }}
                        contentContainerStyle={{
                            paddingBottom:70
                        }}
                    />
                    ):(
                    <View style={{
                        flex:1,
                        justifyContent:"center",
                        alignItems:"center",
                    }}>
                        <Text>
                        No Item Found
                        </Text>
                    </View>
                    )
                )
            }
            <TouchableOpacity style={styles.btn} onPress={()=>navigation.navigate("Create")}>
                <Icon name="ios-add" size={40} color={colors.white} />
            </TouchableOpacity>
        </SafeAreaView>
    );
};

const styles = StyleSheet.create({
    container:{
        flex:1,
        backgroundColor:colors.white
    },
    header:{
        paddingVertical:"2%",
        paddingHorizontal:"5%"
    },
    headerText:{
        fontSize:moderateScale(25),
        fontWeight:"bold",
        color:colors.primary
    },
    btn:{
        backgroundColor:colors.primary,
        justifyContent:"center",
        alignItems:"center",
        width:moderateScale(50),
        height:moderateScale(50),
        borderRadius:moderateScale(25),
        position:"absolute",
        bottom:"5%",
        right:"8%"
    },
    card:{
        flexDirection:"row",
        justifyContent:"space-between",
        alignItems:"center",
        borderWidth:1,
        borderColor:colors.primary,
        marginBottom:"5%",
        paddingHorizontal:"5%",
        paddingVertical:"5%"
    },
    itemName:{
        fontSize:moderateScale(18),
        color:"black",
        fontWeight:"600",
    },
    itemPrice:{
        fontSize:moderateScale(25),
        fontWeight:"400",
        color:colors.primary
    },
    itemStock:{
        fontSize:moderateScale(16),
        fontWeight:"400",
    }
});

export default Home;