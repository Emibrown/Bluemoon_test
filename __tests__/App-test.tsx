/**
 * @format
 */

import 'react-native';
import React from 'react';
import Home from '../src/screens/Home';
import Create from '../src/screens/Create';
import Edit from '../src/screens/Edit';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const initialState = {
  inventory: [],
  loading: true,
  name: "Blue Moon"
}
const mockStore = configureStore()
let store

test('snapshots of Home screen', () => {
  store = mockStore(initialState)
  const home = renderer.create(<Provider store={store}><Home  /></Provider>).toJSON();
  expect(home).toMatchSnapshot();
});

test('snapshots of Create screen', () => {
  store = mockStore(initialState)
  const create = renderer.create(<Provider store={store}><Create  /></Provider>).toJSON();
  expect(create).toMatchSnapshot();
});

test('snapshots of Edit screen', () => {
  store = mockStore(initialState)
  const params = jest.fn();
  const edit = renderer.create(<Provider store={store}><Edit route = {{params}}  /></Provider>).toJSON();
  expect(edit).toMatchSnapshot();
});

