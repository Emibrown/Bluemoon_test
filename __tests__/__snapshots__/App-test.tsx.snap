// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots of Create screen 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": "5%",
        "paddingVertical": "2%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 40,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#509BC7",
          "fontSize": 34.57142857142857,
          "fontWeight": "bold",
        }
      }
    >
      Add Inventory
    </Text>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "marginHorizontal": "5%",
            "marginTop": "5%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25.142857142857142,
              "fontWeight": "400",
            }
          }
        >
          Name
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "height": 70.71428571428572,
              "paddingLeft": "5%",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": "5%",
            "marginTop": "5%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25.142857142857142,
              "fontWeight": "400",
            }
          }
        >
          Total stock
        </Text>
        <TextInput
          keyboardType="numeric"
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "height": 70.71428571428572,
              "paddingLeft": "5%",
            }
          }
          value="0"
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": "5%",
            "marginTop": "5%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25.142857142857142,
              "fontWeight": "400",
            }
          }
        >
          Price
        </Text>
        <TextInput
          keyboardType="numeric"
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "height": 70.71428571428572,
              "paddingLeft": "5%",
            }
          }
          value="0"
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": "5%",
            "marginTop": "5%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25.142857142857142,
              "fontWeight": "400",
            }
          }
        >
          Description
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "height": 70.71428571428572,
              "paddingLeft": "5%",
            }
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#509BC7",
            "justifyContent": "center",
            "marginHorizontal": "5%",
            "marginTop": "10%",
            "opacity": 1,
            "paddingVertical": "4%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25.142857142857142,
            }
          }
        >
          Add Item
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`snapshots of Edit screen 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": "5%",
        "paddingVertical": "2%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 40,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#509BC7",
          "fontSize": 34.57142857142857,
          "fontWeight": "bold",
        }
      }
    >
      Edit Inventory
    </Text>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "marginHorizontal": "5%",
            "marginTop": "5%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25.142857142857142,
              "fontWeight": "400",
            }
          }
        >
          Name
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "height": 70.71428571428572,
              "paddingLeft": "5%",
            }
          }
          value="mockConstructor"
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": "5%",
            "marginTop": "5%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25.142857142857142,
              "fontWeight": "400",
            }
          }
        >
          Total stock
        </Text>
        <TextInput
          keyboardType="numeric"
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "height": 70.71428571428572,
              "paddingLeft": "5%",
            }
          }
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": "5%",
            "marginTop": "5%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25.142857142857142,
              "fontWeight": "400",
            }
          }
        >
          Price
        </Text>
        <TextInput
          keyboardType="numeric"
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "height": 70.71428571428572,
              "paddingLeft": "5%",
            }
          }
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": "5%",
            "marginTop": "5%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25.142857142857142,
              "fontWeight": "400",
            }
          }
        >
          Description
        </Text>
        <TextInput
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "height": 70.71428571428572,
              "paddingLeft": "5%",
            }
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#509BC7",
            "justifyContent": "center",
            "marginHorizontal": "5%",
            "marginTop": "10%",
            "opacity": 1,
            "paddingVertical": "4%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25.142857142857142,
            }
          }
        >
          Add Item
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#e57373",
            "justifyContent": "center",
            "marginHorizontal": "5%",
            "marginTop": "10%",
            "opacity": 1,
            "paddingVertical": "4%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25.142857142857142,
            }
          }
        >
          Delete Item
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`snapshots of Home screen 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
  testID="2"
>
  <View
    style={
      Object {
        "paddingHorizontal": "5%",
        "paddingVertical": "2%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#509BC7",
          "fontSize": 39.285714285714285,
          "fontWeight": "bold",
        }
      }
    >
      Bluemoon
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "justifyContent": "flex-start",
          "paddingTop": "10%",
        },
      ]
    }
  >
    <ActivityIndicator
      color="#509BC7"
      size={62.857142857142854}
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#509BC7",
        "borderRadius": 39.285714285714285,
        "bottom": "5%",
        "height": 78.57142857142857,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": "8%",
        "width": 78.57142857142857,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 40,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</RNCSafeAreaView>
`;
